diff -r 9a65b2bd4e01 src/video/x11/SDL_x11window.c
--- a/src/video/x11/SDL_x11window.c	Mon Jun 04 12:31:14 2012 -0400
+++ b/src/video/x11/SDL_x11window.c	Fri Jun 08 10:36:49 2012 +0200
@@ -762,7 +762,7 @@
 
          XGetWMNormalHints(display, data->xwindow, sizehints, &userhints);
 
-         sizehints->min_width = sizehints->max_height = window->w;
+         sizehints->min_width = sizehints->max_width = window->w;
          sizehints->min_height = sizehints->max_height = window->h;
 
          XSetWMNormalHints(display, data->xwindow, sizehints);
