diff -Naur a/configure.in b/configure.in
--- a/configure.in	2012-09-10 17:03:40.000028000 +0200
+++ b/configure.in	2012-09-11 10:54:04.004562000 +0200
@@ -14,7 +14,7 @@
 
 MAJOR_VERSION=0
 MINOR_VERSION=1
-MICRO_VERSION=2
+MICRO_VERSION=3
 INTERFACE_AGE=0
 BINARY_AGE=1
 VERSION=$MAJOR_VERSION.$MINOR_VERSION.$MICRO_VERSION
@@ -79,15 +79,38 @@
 CFLAGS="$CFLAGS $PANGOFT2_CFLAGS"
 LIBS="$LIBS $PANGOFT2_LIBS"
 
-# Check for SDL
-
-SDL_VERSION=1.2.4
-AM_PATH_SDL($SDL_VERSION,
-            :,
-	    AC_MSG_ERROR([*** SDL version $SDL_VERSION not found!])
-)
-CFLAGS="$CFLAGS $SDL_CFLAGS"
-LIBS="$LIBS $SDL_LIBS"
+# Check for SDL, use: --with-sdl2 for SDL2 
+set -x
+test_sdl2=false
+AC_MSG_CHECKING(for SDL 2)
+AC_ARG_WITH(sdl2,
+     AS_HELP_STRING([--with-sdl2], [Enable SDL 2]),
+     [
+     	if test "x$withval" = "xyes"; then
+			test_sdl2=true	
+		  fi
+     ])
+if test "x$test_sdl2" = "xtrue"; then
+    export PKG_CONFIG_PATH=$prefix/lib/pkgconfig:$PKG_CONFIG_PATH
+	PKG_CHECK_MODULES(SDL2,[sdl2 >= 2.0.0], test_sdl2=true, test_sdl2=false)
+	if test x$test_sdl2 = xfalse; then
+	   AC_MSG_WARN([SDL2 not found])
+	else
+		CFLAGS="$CFLAGS $SDL2_CFLAGS -DWITH_SDL2"
+		LIBS="$LIBS $SDL2_LIBS"
+	fi
+else
+		# Check for SDL
+
+		SDL_VERSION=1.2.4
+		AM_PATH_SDL($SDL_VERSION,
+				    :,
+	    			AC_MSG_ERROR([*** SDL version $SDL_VERSION not found!])
+					)
+		CFLAGS="$CFLAGS $SDL_CFLAGS"
+		LIBS="$LIBS $SDL_LIBS"
+fi
+set +x
 
 AC_CONFIG_FILES([Makefile src/Makefile SDL_Pango.pc docs/Makefile docs/Doxyfile VisualC2003/Makefile Wix/Makefile Wix/merge_module.xml Wix/dev.xml Wix/testbench.xml])
 
diff -Naur a/INSTALL b/INSTALL
--- a/INSTALL	2012-09-10 17:09:48.004652000 +0200
+++ b/INSTALL	2012-09-10 17:10:02.003057000 +0200
@@ -1,8 +1,13 @@
-How to apply SDL_Pango-0.1.2-API-Changes.patch:
------------------------------------------------
+--------------------------------------------------
 Download original source from: http://sourceforge.net/projects/sdlpango/files/latest/download
 cd SDL_Pange-0.1.2/; patch -p1 < SDL_Pango-0.1.2-API-Changes.patch
 
+Then apply SDL_Pango-0.1.2-API-Changes.patch
+
+Next apply SDL_Pango-0.1.2-SDL2-Changes.patch
+Use: autoreconf; ./configure --with-sdl2; make install
+--------------------------------------------------
+-
 To compile and install SDL_Pango:
 
 	In Un*x, run 'configure; make; make install'.
diff -Naur a/README b/README
--- a/README	2012-09-10 21:20:26.002706000 +0200
+++ b/README	2012-09-10 17:05:19.000663000 +0200
@@ -18,3 +18,5 @@
 - bug fix (no global initialization in SDL_Patch.h)
 - typedef SDLPango_Alignment;
 - add 2 functions: SDLPango_CreateContext_GivenFontDesc() and SDLPango_SetText_GivenAlignment()
+
+Patched for compatibilty with SDL2.
diff -Naur a/src/SDL_Pango.c b/src/SDL_Pango.c
--- a/src/SDL_Pango.c	2012-09-10 17:03:12.001397000 +0200
+++ b/src/SDL_Pango.c	2012-09-10 17:22:01.002365000 +0200
@@ -741,8 +741,14 @@
     context->font_desc = pango_font_description_from_string(font_desc);
 
     context->layout = pango_layout_new (context->context);
+#ifdef WITH_SDL2
+	// flags are not used anymore by SDL_CreateRGBSurface
+    SDLPango_SetSurfaceCreateArgs(context, 0, DEFAULT_DEPTH,
+	DEFAULT_RMASK, DEFAULT_GMASK, DEFAULT_BMASK, DEFAULT_AMASK);
+#else // ! WITH_SDL2
     SDLPango_SetSurfaceCreateArgs(context, SDL_SWSURFACE | SDL_SRCALPHA, DEFAULT_DEPTH,
 	DEFAULT_RMASK, DEFAULT_GMASK, DEFAULT_BMASK, DEFAULT_AMASK);
+#endif // ! WITH_SDL2
 
     context->tmp_ftbitmap = NULL;
 
